phpBB/phpbb/template/twig/extension/routing.php

```diff
@@ -31,6 +35,17 @@ public function __construct(\phpbb\routing\helper $helper)
 		$this->helper = $helper;
 	}
 
+	/**
+	* {@inheritdoc}
+	*/
+	public function getFunctions(): array
+	{
+		return [
+			new TwigFunction('url', [$this, 'getUrl'], ['is_safe_callback' => [$this, 'isUrlGenerationSafe']]),
```

Comment:
Are these needed for the symfony upgraded or is this rather something for the twig update? In the latter case, I'd opt to maybe move this to 3.x and discuss the purpose of it there.

Reply:
It would be interesting to have it in 3.3. IMHO.

Reply:
This is needed for Symfony 5 upgrade. The problem here is that in Symfony 5 class `class RoutingExtension` has became final, so phpBB's routing extension can't inherit it but `AbstractExtension`, while the latter doesn't have `function isUrlGenerationSafe`, so phpBB template engine fails.
Hence the code has been borrowed from `class RoutingExtension` to phpBB's twig extension to work.